devicetypes/plaidsystems/spruce-controller.src/spruce-controller.groovy

```diff
@@ -14,476 +10,491 @@
+ * added Pump/Master
+ * add "Dimmer" to Spruce zone child for manual duration
+ 
+**/
+ 
+private def getVERSION() { return "v3.1 10-2020" }
+ 
 metadata {
-	definition (name: 'Spruce Controller', namespace: 'plaidsystems', author: 'Plaid Systems') {
-		capability 'Switch'
-        capability 'Configuration'
-        capability 'Refresh'
-        capability 'Actuator'
-        capability 'Valve'        
-		
-        attribute 'switch', 'string'
-        attribute 'switch1', 'string'
-		attribute 'switch2', 'string'
-		attribute 'switch8', 'string'
-		attribute 'switch5', 'string'
-		attribute 'switch3', 'string'
-		attribute 'switch4', 'string'
-		attribute 'switch6', 'string'
-		attribute 'switch7', 'string'
-        attribute 'switch9', 'string'
-		attribute 'switch10', 'string'
-		attribute 'switch11', 'string'
-		attribute 'switch12', 'string'
-		attribute 'switch13', 'string'
-		attribute 'switch14', 'string'
-		attribute 'switch15', 'string'
-        attribute 'switch16', 'string'
-		attribute 'rainsensor', 'string'
-        attribute 'status', 'string'
-        attribute 'tileMessage', 'string'
-        attribute 'minutes', 'string'
-        attribute 'VALUE_UP', 'string'
-        attribute 'VALUE_DOWN', 'string'        
-        
-        command 'levelUp'
-        command 'levelDown'
-        command 'programOn'
-        command 'programOff'
-        command 'programWait'
-        command 'programEnd'
-        
-        command 'on'
-        command 'off'
-        command 'zon'
-        command 'zoff'
-        command 'z1on'
-		command 'z1off'
-		command 'z2on'
-		command 'z2off'        
-		command 'z3on'
-		command 'z3off'
-		command 'z4on'
-		command 'z4off'
-		command 'z5on'
-		command 'z5off'
-		command 'z6on'
-		command 'z6off'
-		command 'z7on'
-		command 'z7off'
-		command 'z8on'
-		command 'z8off'
-        command 'z9on'
-        command 'z9off'
-		command 'z10on'
-		command 'z10off'
-		command 'z11on'
-		command 'z11off'
-		command 'z12on'
-		command 'z12off'
-		command 'z13on'
-		command 'z13off'
-		command 'z14on'
-		command 'z14off'
-		command 'z15on'
-		command 'z15off'
-		command 'z16on'
-		command 'z16off'
-        
-        command 'config'
-        command 'refresh'        
-        command 'rain'
-        command 'manual'
-        command 'manualTime'
-        command 'settingsMap'
-        command 'writeTime'
-        command 'writeType'        
-        command 'notify'
-        command 'updated'      
-        
-		//ST release
-		//fingerprint endpointId: '1,2,3,4,5,6,7,8,9,10,11,12,13,14,15,16,17,18', profileId: '0104', deviceId: '0002', deviceVersion: '00', inClusters: '0000,0003,0004,0005,0006,000F', outClusters: '0003, 0019', manufacturer: 'PLAID SYSTEMS', model: 'PS-SPRZ16-01'
+	definition (name: "Spruce Controller", namespace: "plaidsystems", author: "Plaid Systems", mnmn: "SmartThingsCommunity", vid: "566e4d4f-93bc-3c86-b7e1-8a4e212683b0"){//vid: "5e1426e6-1943-3331-8880-6c3e7743f2cc"){
```

Comment:
Can you give me insights how did you defined this vid? Did you used SmartThings CLI for this?

Reply:
Yes, SmartThings CLI.  Newest commit has an updated vid with new presentation, because previous presentation had a duplicate rainSensor component I had put in to try and force a presentation update previously.